misc/scripts/docs_deploybot/deploy_docs.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2022, GNU LGPL, Obspy developers
+from datetime import datetime, timezone
+from io import BytesIO
+import logging
+import os
+import shutil
+import tarfile
+from zipfile import ZipFile
+
+import requests
+
+
+logging.basicConfig(level=logging.INFO, format='%(message)s')
+log = logging.getLogger('deploy_docs')
+TOKEN = os.getenv('GITHUB_TOKEN', '')
+RKW = {'headers': {'accept': 'application/vnd.github.v3+json',
+                   'authorization': f'token {TOKEN}'}}
+API = 'https://api.github.com/repos/obspy/obspy/'
+BASE = '/home/obspy/htdocs/docs/'
+PATH = {'pr': BASE + 'pr/{pr}/',
+        'master': BASE + 'master/',
+        'stable': BASE + 'stable/',
+        'archive': BASE + 'archive/{v}/'}
+URL = 'https://docs.obspy.org/pr/{pr}'
```

Comment:
is it normal that the URL is not built from the PATH? (or only for pr) ?

Reply:
Yes that is normal, it can be configured. The advantage is that the stable branch sits in an extra dictionary and not in root.